libs/bits/bit_array.go

```diff
@@ -182,11 +155,24 @@ func (bA *BitArray) Or(o *BitArray) *BitArray {
 	}
 	bA.mtx.Lock()
 	o.mtx.Lock()
-	c := bA.copyBits(cmtmath.MaxInt(bA.Bits, o.Bits))
-	smaller := cmtmath.MinInt(len(bA.Elems), len(o.Elems))
-	for i := 0; i < smaller; i++ {
```

Comment:
there was a bug in the previous implementation that would stop iterating if `o` had more elements than `bA` so I fixed it and then `or` passed all the unit tests for `TestAddBitArray` so I removed that method which mutated the caller and instead kept `Or` which returns a new array.

Reply:
I can relate to your concern around accidentally modifying behavior for some upstream code that isn't covered in unit tests.

I wish upstream code had enough unit test coverage that we didn't have to be paranoid about making changes, but I'll revert all my changes and then try to fix the deadlock in a different way. I don't have a real strategy for that yet so may take some time.